### PR TITLE
Update the WP packages with fixes prior to WP 6.2 beta 5

### PR DESCRIPTION
**Pull request by ntsekouras**
Trac ticket: https://core.trac.wordpress.org/ticket/57471

This updated the packages to the latest versions with the latest fixes.

https://github.com/WordPress/gutenberg/commit/d14fea64bd9ad9d2d86047936974dbed39a5b5b3

Includes the following changes:

- Add border radius to off canvas navigation menu items
- Remove border from quick inserter child elements
- Show variant patterns even if there are no patterns for the Query Loop block
- Order initial block items in Navigation with PrivateInserter
- Update: Add descriptions to all panels in the Site Editor's dark side
- Fix typo in template parts description
- Fix: Browse mode descriptions margin
- Fix: Show creation popover on empty page links in the navigation sidebar
- Make sure the directly inserted block in the Nav block is a Page link
- Fix browser history when synchronising state with urls
- Navigation Sidebar: Change the logic about which navigation gets selected for the sidebar
- Fixes extra UI in navigation block inspector
- Renames parent selection boolean param and improves docs
- Widget Importer: Fix Widget Group block imports
- Don't add Post Content layout styles to title in the post editor
- Site editor: Add hover animation to site editor canvas
- Prevent the saving button from showing when renaming templates
- Navigation Block: Fix big spinner
- Navigation: Don't save the level of the link in an attribute
- Focus 1st parent block on block remove, if no previous block is available
- Navigation: Performance: improve params in `block_core_navigation_get_most_recently_published_navigation()`
- Navigation Block inspector: fix link UI popover opening on links that have a url
- Fix for `WP_Theme_JSON_Resolver::get_merged_data()`
- Site Editor: Add page details when viewing a specific page.
- Fix site editor sidebar scrollbars
- Fix: Custom link UI does appears outside canvas on the sidebar navigation
- Fix: Make navigation page list load its items on navigation sidebar

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]
Tested critical fixes like Site Editor URL history bug and Navigation management updates. The fixes work as expected.

File: src/wp-includes/blocks/navigation-link.php [hunk 1]
```diff
@@ -9,18 +9,17 @@
...
  *
- * @param  array $context    Navigation block context.
- * @param  array $attributes Block attributes.
+ * @param  array $context     Navigation block context.
+ * @param  array $attributes  Block attributes.
+ * @param  bool  $is_sub_menu Whether the link is part of a sub-menu.
  * @return array Colors CSS classes and inline styles.
  */
-function block_core_navigation_link_build_css_colors( $context, $attributes ) {
+function block_core_navigation_link_build_css_colors( $context, $attributes, $is_sub_menu = false ) {
```
Comment (hellofromtonya) [inline src/wp-includes/blocks/navigation-link.php:17]: Changes from https://github.com/WordPress/gutenberg/commit/dbe47a5e8f829f4a21b9d7d580098ecefb73447a#diff-96cfb99105116cdbf48eface8e7ac4dff062da11f291996bf5748d5a8b90f3e0.

File: src/wp-includes/blocks/navigation-submenu.php [hunk 2]
```diff
@@ -9,18 +9,17 @@
...
  *
- * @param  array $context    Navigation block context.
- * @param  array $attributes Block attributes.
+ * @param  array $context     Navigation block context.
+ * @param  array $attributes  Block attributes.
+ * @param  bool  $is_sub_menu Whether the block is a sub-menu.
  * @return array Colors CSS classes and inline styles.
  */
-function block_core_navigation_submenu_build_css_colors( $context, $attributes ) {
+function block_core_navigation_submenu_build_css_colors( $context, $attributes, $is_sub_menu = false ) {
```
Comment (hellofromtonya) [inline src/wp-includes/blocks/navigation-submenu.php:17]: Changes to this function are from https://github.com/WordPress/gutenberg/commit/dbe47a5e8f829f4a21b9d7d580098ecefb73447a#diff-a22b96046ab15d77ae0b88e24a4cf01e5dbfea45c5bb8e489db9204c134026bf.

File: src/wp-includes/blocks/navigation.php [hunk 3]
```diff
@@ -373,12 +373,14 @@ function block_core_navigation_get_most_recently_published_navigation() {
 
 	// Default to the most recently created menu.
 	$parsed_args = array(
-		'post_type'      => 'wp_navigation',
-		'no_found_rows'  => true,
-		'order'          => 'DESC',
-		'orderby'        => 'date',
-		'post_status'    => 'publish',
-		'posts_per_page' => 1, // get only the most recent.
+		'post_type'              => 'wp_navigation',
```
Comment (hellofromtonya) [inline src/wp-includes/blocks/navigation.php:376]: Changes here are from https://github.com/WordPress/gutenberg/commit/ea74a306e5289a90d29c0b3eb1d9b4ef893f7f19.

**Comment by hellofromtonya** [comment]
# Test Report

Issue: [Site Editor] White screen when using browser back button https://github.com/WordPress/gutenberg/issues/48726

## Steps to Reproduce

1. Navigate to the Site Editor 🐞
2. Click on the navigation sidebar.
3. Click the browser's back button 🐞
4. Click the browser's back button again 🐞

## Expected Results

When testing a patch to validate it works as expected:
- ✅ The Site Editor's URL should be `http://localhost:8889/wp-admin/site-editor.php`
- ✅ Clicking on the browser's back button should return to the Dashboard `http://localhost:8889/wp-admin/`

When reproducing a bug:
- ❌ When entering the Site Editor, the `path` query arg is wrong `/site-editor.php?canvas=init&path=%2F` 
- ❌ First click > nothing happens
- ❌ Second click > Get a white screen 

=== Environment

* OS: macOS
* Browser: Chrome, Firefox, and Edge
* Plugins: none
* Theme: TT3
* WordPress: trunk and with this PR

==== Actual Results

When reproducing a bug/defect:
- ❌ Confirmed: the Site Editor URL `path` query arg is wrong `/site-editor.php?canvas=init&path=%2F`.
- ❌ Confirmed: clicking on the browser's back button does nothing.
- ❌ Confirmed: again clicking on the browser's back button shows a white screen.

When testing the bugfix patch:
- ✅ Confirmed: Site Editor's URL is correct.
- ✅ Confirmed: clicking on the browser's back button does return to the Dashboard.

== Summary
- Before this patch: Able to reproduce the issue ✅
- After this patch: issue is resolved ✅

**Review by hellofromtonya — APPROVED** [review]
* Confirmed package version numbers match https://github.com/WordPress/gutenberg/commit/d14fea64bd9ad9d2d86047936974dbed39a5b5b3 ✅
* Confirmed fix in `navigation-link.php` matches [change in Gutenberg](https://github.com/WordPress/gutenberg/commit/dbe47a5e8f829f4a21b9d7d580098ecefb73447a#diff-96cfb99105116cdbf48eface8e7ac4dff062da11f291996bf5748d5a8b90f3e0) ✅ 
* Confirmed fix in `navigation-submenu.php` matches [change in Gutenberg](https://github.com/WordPress/gutenberg/commit/dbe47a5e8f829f4a21b9d7d580098ecefb73447a#diff-a22b96046ab15d77ae0b88e24a4cf01e5dbfea45c5bb8e489db9204c134026bf) ✅ 
* Confirmed fix in `navigation.php` matches [change in Gutenberg](https://github.com/WordPress/gutenberg/commit/ea74a306e5289a90d29c0b3eb1d9b4ef893f7f19) ✅ 
* Confirmed the browser's back button in Site Editor bug is fixed ([see test report](https://github.com/WordPress/wordpress-develop/pull/4182#issuecomment-1458345757)) ✅ 

Ready for commit 👍

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/55475.

**Pull request closed without merging**